### PR TITLE
Migrate RegistrationRequest to SQL (2 of 4)

### PR DESCRIPTION
**Pull request by orangejenny**
Branched off of https://github.com/dimagi/commcare-hq/pull/26555, first new commit is the first one with "PR2".

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-1015) (same as previous PR)

### CODE REVIEW

**Review by biyeun — APPROVED** [review]

File: corehq/apps/cleanup/deletable_doc_types.py [hunk 1]
```diff
@@ -23,5 +23,6 @@
     'ForwardingRule': (MAIN_DB,),
     'ForwardingRule-Deleted': (MAIN_DB,),
     'GlobalAppConfig': (settings.NEW_APPS_DB,),
+    'RegistrationRequest': (MAIN_DB,),
```
Comment (snopoke) [inline corehq/apps/cleanup/deletable_doc_types.py]: perhaps done later but we should also update the domain deletion command to remove the SQL data

**Review by snopoke — APPROVED** [review]

File: corehq/apps/cleanup/deletable_doc_types.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/cleanup/deletable_doc_types.py]: 52983602ba6 (in https://github.com/dimagi/commcare-hq/pull/26556)

**Review by millerdev — APPROVED** [review]

**Comment by orangejenny** [comment]
Rebased to incorporate changes in previous PR, no need to re-review.

**Comment by orangejenny** [comment]
Updated to revert a946a89edaa59a93dafe5e0b3cf9f247741e53ff which added `RegistrationRequest` to `DELETABLE_COUCH_DOC_TYPES` which shouldn't be done until the cleanup PR, and also to increment the migration to account for https://github.com/dimagi/commcare-hq/pull/26660

**Review by millerdev — APPROVED** [review]

**Pull request merged**